src/Pact/Analyze/Types/Languages.hs

```diff
@@ -247,6 +264,50 @@ data Prop a
   | CoreProp     (Core Prop a)
   deriving (Show, Eq)
 
+instance UserShow a => UserShow (PropSpecific a) where
+  userShowsPrec _d = \case
+    Abort                   -> STransactionAborts
+    Success                 -> STransactionSucceeds
+    Result                  -> SFunctionResult
+    Forall _ var ty x       -> parenList
+      [SUniversalQuantification, parens (var <> ":" <> userShow ty), userShow x]
+    Exists _ var ty x       -> parenList
+      [SExistentialQuantification, parens (var <> ":" <> userShow ty), userShow x]
+    TableWrite tab          -> parenList [STableWritten, userShow tab]
+    TableRead  tab          -> parenList [STableRead, userShow tab]
+    ColumnWrite tab col     -> parenList ["column-write", userShow tab, userShow col]
+    ColumnRead tab col      -> parenList ["column-read", userShow tab, userShow col]
+    IntCellDelta tab col rk -> parenList [SCellDelta, userShow tab, userShow col, userShow rk]
+    DecCellDelta tab col rk -> parenList [SCellDelta, userShow tab, userShow col, userShow rk]
+    IntColumnDelta tab col  -> parenList [SColumnDelta, userShow tab, userShow col]
+    DecColumnDelta tab col  -> parenList [SColumnDelta, userShow tab, userShow col]
+    RowRead tab rk          -> parenList [SRowRead, userShow tab, userShow rk]
+    RowReadCount tab rk     -> parenList [SRowReadCount, userShow tab, userShow rk]
+    RowWrite tab rk         -> parenList [SRowWritten, userShow tab, userShow rk]
+    RowWriteCount tab rk    -> parenList [SRowWriteCount, userShow tab, userShow rk]
+
+instance UserShow a => UserShow (Prop a) where
+  userShowsPrec d = \case
+    PropSpecific p -> userShowsPrec d p
+    CoreProp     p -> userShowsPrec d p
+
+instance UserShow a => UserShow (Core Prop a) where
```

Comment:
Unfortunately this is duplicated (search for `UserShow (Core Term a)`). I need to look into this further.